src/EnergyPlus/DXCoils.cc

```diff
@@ -2979,895 +2979,921 @@ void GetDXCoils(EnergyPlusData &state)
 
     // Loop over the Pumped DX Water Heater Coils and get & load the data
     CurrentModuleObject = HVAC::cAllCoilTypes(HVAC::CoilDX_HeatPumpWaterHeaterPumped);
-    for (DXHPWaterHeaterCoilNum = 1; DXHPWaterHeaterCoilNum <= state.dataDXCoils->NumDXHeatPumpWaterHeaterPumpedCoils; ++DXHPWaterHeaterCoilNum) {
+    auto &s_ip = state.dataInputProcessing->inputProcessor;
+    auto const instances_whPumped = s_ip->epJSON.find(CurrentModuleObject);
+    if (instances_whPumped != s_ip->epJSON.end()) {
+        std::string cFieldName;
+        auto const &schemaProps = s_ip->getObjectSchemaProps(state, CurrentModuleObject);
+        auto &instancesValue = instances_whPumped.value();
+        for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+
+            ++DXCoilNum;
+            auto const &fields = instance.value();
+            std::string const &thisObjectName = instance.key();
+
+            // allocate single performance mode for numeric field strings used for sizing routine
+            state.dataDXCoils->DXCoilNumericFields(DXCoilNum).PerfMode.allocate(1);
+            state.dataDXCoils->DXCoilNumericFields(DXCoilNum).PerfMode(1).FieldNames.allocate(MaxNumbers);
+            state.dataDXCoils->DXCoilNumericFields(DXCoilNum).PerfMode(1).FieldNames = cNumericFields;
```

Comment:
The `Coil:WaterHeating:AirToWaterHeatPump:Pumped` and `Coil:WaterHeating:AirToWaterHeatPump:Wrapped`  the JSON format.